hostpath-provisioner.go

```diff
@@ -99,54 +101,59 @@ func (p *hostPathProvisioner) Provision(options controller.ProvisionOptions) (*v
 		return nil, &controller.IgnoredError{Reason: "identity annotation on pvc does not match ours"}
 	}
 	vPath := path.Join(p.pvDir, options.PVName)
+	pvCapacity, err := calculatePvCapacity(p.pvDir)
 	if p.useNamingPrefix {
 		vPath = path.Join(p.pvDir, options.PVC.Name+"-"+options.PVName)
 	}
-	glog.Infof("creating backing directory: %v", vPath)
 
-	if err := os.MkdirAll(vPath, 0777); err != nil {
-		return nil, err
-	}
+	if pvCapacity != nil {
```

Comment:
What happens when the user requests a PVC greater than `pvCapacity`?

Reply:
Since the capacity is the max disk size, if the user requests a PVC larger than that, it cannot be satisfied and the PVC will remain pending, however we will create a PV, which might not be what we want.

Reply:
#5 fixes this.